src/en/mangafun/src/eu/kanade/tachiyomi/extension/en/mangafun/MangaFun.kt

```diff
@@ -0,0 +1,295 @@
+                        else -> throw IllegalStateException("Unhandled sort option")
+                    }
+
+                    if (filter.state?.ascending != true) {
+                        directory = directory.reversed()
+                    }
+                }
+                else -> {}
+            }
+        }
+
+        return parseDirectory(1)
+    }
+
+    override fun getMangaUrl(manga: SManga) = "$baseUrl${manga.url}"
+
+    override fun mangaDetailsRequest(manga: SManga): Request {
+        val slug = manga.url.substringAfterLast("/")
+        val nextDataUrl = "$baseUrl/_next/data/$nextBuildId/title/$slug.json"
+
+        return GET(nextDataUrl, headers)
+    }
+
+    override fun mangaDetailsParse(response: Response): SManga {
+        val data = response.parseAs<NextPagePropsWrapperDto>()
+            .pageProps
+            .dehydratedState
+            .queries
+            .first()
+            .state
+            .data
+
+        return json.decodeFromJsonElement<MangaDto>(data).toSManga()
+    }
+
+    override fun chapterListRequest(manga: SManga) = mangaDetailsRequest(manga)
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val data = response.parseAs<NextPagePropsWrapperDto>()
+            .pageProps
+            .dehydratedState
+            .queries
+            .first()
+            .state
+            .data
+
+        val mangaData = json.decodeFromJsonElement<MangaDto>(data)
+        return mangaData.chapters.map { it.toSChapter(mangaData.id, mangaData.name) }.reversed()
+    }
+
+    override fun getChapterUrl(chapter: SChapter) = "$baseUrl${chapter.url}"
+
+    override fun pageListRequest(chapter: SChapter): Request {
+        val chapterId = chapter.url.substringAfterLast("/").substringBefore("-")
+
+        return GET("$apiUrl/chapter/$chapterId", headers)
+    }
+
+    override fun pageListParse(response: Response): List<Page> {
+        val encoded = Base64.encode(response.body.bytes(), Base64.DEFAULT or Base64.NO_WRAP).toString(Charsets.UTF_8)
+        val decoded = LZString.decompressFromBase64(encoded)
+        val compressedJson = json.parseToJsonElement(decoded).jsonArray
+        val decompressedJson = DecompressJson.decompress(compressedJson).jsonObject
+
+        Log.d("MangaFun", Json.encodeToString(decompressedJson))
+
+        return decompressedJson.jsonObject["p"]!!.jsonArray.mapIndexed { i, it ->
+            Page(i, imageUrl = MangaFunUtils.getImageUrlFromHash(it.jsonArray[0].jsonPrimitive.content))
+        }
+    }
+
+    override fun imageUrlParse(response: Response) = throw UnsupportedOperationException()
+
+    override fun getFilterList() = FilterList(
+        GenreFilter(),
+        TypeFilter(),
+        StatusFilter(),
+        SortFilter(),
+    )
+
+    private fun parseDirectory(page: Int): MangasPage {
+        val endRange = (page * 24).let { if (it < directory.size) it else directory.size }
```

Comment:
```suggestion
        val endRange = min((page * 24), directory.size)
```
import `kotlin.math.min`